### ISSUE TITLE
Axe Options RunOnlyObject not being recognized in Angular 6 project.

### ISSUE THREAD

**Issue by Bmahecha123** [dequelabs/axe-core#894]
Hello everyone,
When I am running Axe.Run(ElementContext, Options): Promise<AxeResults> I am getting the following error in my Angular test using Jasmine. 

```
Error: runOnly.values must be a non-empty array
```

Checking the Axe Core code this object should be recognized but it is not for whatever reason. Any help with this would be greatly appreciated.

```
interface RunOnly {
		type: RunOnlyType,
		values?: TagValue[] | RunOnlyObject
	}
	interface RunOptions {
		runOnly?: RunOnly,
		rules?: Object,
		iframes?: Boolean,
		elementRef?: Boolean,
		selectors?: Boolean
	}
```

Here is my test case:

<pre><code>
axe-core version: 3.0.2

For Tooling issues:
- Node version: v9.8.0
- Platform:  Mac

it('should have no a11y violations', async() => {

    const rules: Axe.RunOnlyObject = {
      include: ['wcag2a', 'wcag2aa'],
      exclude: ['section508']
    };

    const options: Axe.RunOptions = {
      runOnly: {
        type: 'tags',
        values: rules
      }
    }
   
    let axeTest = await Axe.run(fixture.debugElement.nativeElement, options);
    console.dir(axeTest.violations);
    expect(axeTest.violations.length).toEqual(3);
  });
</pre></code>

**Comment by marcysutton** [comment]
Ah yes, I see that now. I'll reopen it. @WilcoFiers looks like we forgot an API option in #774

**Comment by jkodu** [comment]
@Bmahecha123 

I will revert with an update. 
In the interim, if you could extend the `axe` interface to the type it is expecting should help with the failure.

**Comment by WilcoFiers** [comment]
I think it needs to be this:

```ts
    type ContextObject = {
		include?: string[] | string[][];
		exclude?: string[] | string[][];
	};

	type RunCallback = (error: Error, results: AxeResults) => void;

	type ElementContext = Node | string | ContextObject;

	interface RunOnly {
		type: RunOnlyType;
		values: TagValue[] | string[];
	}
```

- Renamed RunOnlyObject to ContextObject.
- Removed RunOnlyObject from `RunOnly` that doesn't take an object with include/exclude.
- Made `values` required. Axe complains if you don't set it.

**Comment by Bmahecha123** [comment]
Thank you for the reply; I will experiment with this as soon as I can!

**Comment by padmavemulapati** [comment]
Dev task
